### PR TITLE
Update SortCommand for v1.3

### PR DESCRIPTION
**Pull request by JordanChua**
- Added `SortProjectStatusCommand` and `SortSpecialisationCommand`
- Updated DG for these 2 commands
- Resolved issue that the list is permanently sorted once `SortCommand` is run :)))))
- Need to update DG seqeunce diagrams which will be done later cuz i nid sleep :(
- Resolves issue #107, #22 and #31

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -470,6 +470,41 @@ The following sequence diagram shows how the MarkCommand operation works:
     * Pros: More comprehensive search for projects with the required keyword.
     * Cons: Much harder to implement, as it requires a field-less search.
 
+
+### \[Proposed\] `Sort` Feature
```
Comment (RussellDash332) [inline docs/DeveloperGuide.md:474]: This is not a `Proposed` feature, so you might want to remove the word.

File: src/main/java/jeryl/fyp/logic/commands/SortSpecialisationCommand.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+/**
+ * Sorts all projects in the FYP manager by the specialisation, which is equivalent to the project name
+ */
+public class SortSpecialisationCommand extends Command {
+
+    public static final String COMMAND_WORD = "sort-sp";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Sorts the FYP projects "
+            + "by the specialisation of the  projects followed by alphabetical order"
+            + "Note that the format is fixed, hence only " + COMMAND_WORD + " is accepted"
```
Comment (RussellDash332) [inline src/main/java/jeryl/fyp/logic/commands/SortSpecialisationCommand.java:16]: Missing a `\n` right after `accepted`

File: src/main/java/jeryl/fyp/model/FypManager.java [hunk 3]
```diff
@@ -20,6 +20,7 @@ public class FypManager implements ReadOnlyFypManager {
 
     private final UniqueStudentList students;
 
+
```
Comment (RussellDash332) [inline src/main/java/jeryl/fyp/model/FypManager.java:23]: Any reason for this additional newline?

File: src/main/java/jeryl/fyp/logic/commands/SortProjectStatusCommand.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+/**
+ * Sorts projects in the FYP manager by the project status, followed by alphabetical order
+ */
+public class SortProjectStatusCommand extends Command {
+
+    public static final String COMMAND_WORD = "sort-s";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Sorts the FYP projects "
+            + "by the current project status, in the order YTS, IP followed by DONE"
+            + "Note that the format is fixed, hence only " + COMMAND_WORD + " is accepted"
```
Comment (RussellDash332) [inline src/main/java/jeryl/fyp/logic/commands/SortProjectStatusCommand.java:16]: Also missing a `\n` here

File: src/test/java/jeryl/fyp/logic/parser/FypManagerParserTest.java [hunk 5]
```diff
@@ -120,6 +122,20 @@ public void parseCommand_mark() throws Exception {
...
+                instanceof SortProjectStatusCommand);
+    }
+
+    @Test
+    public void parseCommand_sortSpecialisation() throws Exception {
+        assertTrue(parser.parseCommand(SortSpecialisationCommand.COMMAND_WORD) instanceof SortSpecialisationCommand);
+        assertTrue(parser.parseCommand(SortSpecialisationCommand.COMMAND_WORD + " 3")
+                instanceof SortSpecialisationCommand);
+    }
+
```
Comment (RussellDash332) [inline src/test/java/jeryl/fyp/logic/parser/FypManagerParserTest.java:138]: There can be more tests here but I suppose we'll hold this until v1.4.

**Review by RussellDash332 — CHANGES_REQUESTED** [review]
Minor changes needed so less work needs to be done on v1.4. Thanks for the hard work!

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (JordanChua) [reply docs/DeveloperGuide.md:474]: Changed!

File: src/main/java/jeryl/fyp/logic/commands/SortSpecialisationCommand.java (hunk 2 above)
Comment (JordanChua) [reply src/main/java/jeryl/fyp/logic/commands/SortSpecialisationCommand.java:16]: Resolved!

File: src/main/java/jeryl/fyp/model/FypManager.java (hunk 3 above)
Comment (JordanChua) [reply src/main/java/jeryl/fyp/model/FypManager.java:23]: Nope I have removed the additional newline

File: src/test/java/jeryl/fyp/logic/parser/FypManagerParserTest.java (hunk 5 above)
Comment (JordanChua) [reply src/test/java/jeryl/fyp/logic/parser/FypManagerParserTest.java:138]: Sure I'll raise an issue separately!

**Review by RussellDash332 — APPROVED** [review]

**Pull request merged**